docs/migration.md

```diff
@@ -3,198 +3,93 @@ title: Migration
+  Prometheus v2 these combinations didn't match.
+
+| Regex      | Additional matches  |
+| -----      | ------              |
+| ".*"       | "foo\n", "Foo\nBar" |
+| "foo.?bar" | "foo\nbar"          |
+| "foo.+bar" | "foo\nbar"          |
+
+  If you want Prometheus v3 to behave like v2 did, you will have to change your 
+  regular expressions by replacing all `.` patterns with `[^\n]`, e.g.  
+  `foo[^\n]*`.
+- Lookback and range selectors are left open and right closed (previously left 
+  closed and right closed). This change affects queries when the evaluation time 
+  perfectly aligns with the sample timestamps. For example assume querying a 
+  timeseries with even spaced samples exactly 1 minute apart. Before Prometheus 
+  3.x, range query with `5m` will mostly return 5 samples. But if the query 
+  evaluation aligns perfectly with a scrape, it would return 6 samples. In 
+  Prometheus 3.x queries like this will always return 5 samples.
+  This change has likely few effects for everyday use, except for some sub query 
+  use cases.
+  Query front-ends that align queries usually align sub-queries to multiples of 
+  the step size. These sub queries will likely be affected.
+  Tests are more likely to affected. To fix those either adjust the expected 
+  number of samples or extend to range by less then one sample interval.
+- The `holt_winters` function has been renamed to `double_exponential_smoothing` 
+  and is now guarded by the `promql-experimental-functions` feature flag.
+  If you want to keep using holt_winters, you have to do both of these things:
+    - Rename holt_winters to double_exponential_smoothing in your queries.
+    - Pass `--enable-feature=promql-experimental-functions` in your Prometheus 
+      cli invocation..
 
 ## Miscellaneous
 
-### Prometheus non-root user
+### TSDB format and downgrade
+The TSDB format has been changed in Prometheus v2.55 in preparation for changes 
+to the index format. Consequently a Prometheus v3 tsdb can only be read by a 
+Prometheus v2.55 or newer.
 
-The Prometheus Docker image is now built to [run Prometheus
-as a non-root user](https://github.com/prometheus/prometheus/pull/2859). If you
-want the Prometheus UI/API to listen on a low port number (say, port 80), you'll
-need to override it. For Kubernetes, you would use the following YAML:
+### TSDB Storage contract
+TSDB compatible storage is now expected to return results matching the specified 
+selectors. This might impact some third party implementations, most likely 
+implementing `remote_read`.
+This contract is not explicitly enforced, but can cause undefined behavior.
 
-```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  name: security-context-demo-2
-spec:
-  securityContext:
-    runAsUser: 0
-...
-```
+### UTF-8 names
+Prometheus v3 supports UTF-8 in metrics and label names. This means metrics and 
+label names can change after upgrading according to what is exposed by 
+endpoints.
```

Comment:
I need to expand this to explain how to adjust configs to disallow utf8.  Should I do that here or in a subsequent PR?

Reply:
You can either create a PR against https://github.com/jan--f/prometheus/tree/3.0-migration or we merge this soon and then incremental improvements can be parallelized in p/p.